infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/Names.java

```diff
@@ -20,40 +25,37 @@
  * @author Sergii Leshchenko
  */
 public class Names {
+
   static final char WORKSPACE_ID_PREFIX_SEPARATOR = '.';
 
   static final String ROUTE_PREFIX = "route";
   static final int ROUTE_PREFIX_SIZE = 8;
 
-  /** Returns machine name that has the following format `{POD_NAME}/{CONTAINER_NAME}`. */
-  public static String machineName(String podName, String containerName) {
-    return podName + '/' + containerName;
-  }
-
-  /** Returns machine name that has the following format `{POD_NAME}/{CONTAINER_NAME}`. */
-  public static String machineName(Pod pod, Container container) {
-    return machineName(pod.getMetadata().getName(), container.getName());
-  }
-
   /**
-   * Returns pod name that is extracted from machine name.
-   *
-   * @see #machineName(String, String)
+   * Returns machine name that has the following format `{POD_NAME}/{CONTAINER_NAME}` or machine
+   * name from configuration when it's not composite.
    */
-  public static String podName(String machineName) {
-    return machineName.split("/")[0];
+  public static String machineName(Pod pod, Container container) {
```

Comment:
You dont use anything from container except the name. Maybe leave param `containerName`?

Reply:
I'd leave `Container` as it gives us an opportunity to use something else from container later.